File: projects/01-promo-map/flutter_app/lib/data/models/store.dart
import 'discount_brief.dart';

class Store {
  final int id;
  final String name;
  final String brand;
  final String category;
  final String address;
  final double latitude;
  final double longitude;
  final String phone;
  final String iconColor;
  final String iconLetter;
  final double? distanceM;
  final List<DiscountBrief> discounts;

  const Store({
    required this.id,
    required this.name,
    required this.brand,
    this.category = 'general',
    this.address = '',
    required this.latitude,
    required this.longitude,
    this.phone = '',
    this.iconColor = '#FF6B35',
    this.iconLetter = 'S',
    this.distanceM,
    this.discounts = const [],
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: json['id'] as int,
      name: json['name'] as String,
      brand: json['brand'] as String,
      category: (json['category'] as String?) ?? 'general',
      address: (json['address'] as String?) ?? '',
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      phone: (json['phone'] as String?) ?? '',
      iconColor: (json['icon_color'] as String?) ?? '#FF6B35',
      iconLetter: (json['icon_letter'] as String?) ?? 'S',
      distanceM: json['distance_m'] != null
          ? (json['distance_m'] as num).toDouble()
          : null,
      discounts: (json['discounts'] as List<dynamic>?)
              ?.map((e) => DiscountBrief.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}

File: projects/01-promo-map/backend/schemas/store.py
"""매장 스키마"""
from pydantic import BaseModel
from typing import Optional, List

__all__ = [
    "StoreResponse", "StoreDetailResponse", "StoreListResponse",
    "NearbyStoreResponse", "StoreCreateRequest", "StoreUpdateRequest",
]


class DiscountBrief(BaseModel):
    id: int
    type: str
    value: float
    description: str


class StoreResponse(BaseModel):
    id: int
    name: str
    brand: str
    category: str
    address: str
    latitude: float
    longitude: float
    phone: str = ""
    icon_color: str
    icon_letter: str
    distance_m: Optional[float] = None
    discounts: List[DiscountBrief] = []

    class Config:
        from_attributes = True


class StoreDetailResponse(StoreResponse):
    reviews_count: int = 0
    avg_rating: Optional[float] = None
    is_favorited: bool = False


class StoreListResponse(BaseModel):
    count: int
    stores: List[StoreResponse]


class NearbyStoreResponse(BaseModel):
    count: int
    stores: List[StoreResponse]


class StoreCreateRequest(BaseModel):
    name: str
    brand: str
    category: str = "general"
    address: str = ""
    latitude: float
    longitude: float
    phone: str = ""
    icon_color: str = "#FF6B35"
    icon_letter: str = "S"


class StoreUpdateRequest(BaseModel):
    name: Optional[str] = None
    brand: Optional[str] = None
    category: Optional[str] = None
    address: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    phone: Optional[str] = None
    icon_color: Optional[str] = None
    icon_letter: Optional[str] = None
    is_active: Optional[bool] = None
